src/scenarios/shared/androidinstrumentation.py

```diff
@@ -0,0 +1,148 @@
+from performance.constants import UPLOAD_CONTAINER, UPLOAD_STORAGE_URI, UPLOAD_TOKEN_VAR, UPLOAD_QUEUE
+from shared.util import helixuploaddir, uploadtokenpresent, xharnesscommand
+from shared.const import *
+from subprocess import CalledProcessError
+
+class AndroidInstrumentationHelper(object):
+
+    def runtests(self, packagepath: str, packagename: str, instrumentationname: str):
+        '''
+        Runs Android Instrumentation tests.
+        '''
+        try:
+            # Do the actual work here
+            cmdline = xharnesscommand() + ['android', 'state', '--adb']
+            adb = RunCommand(cmdline, verbose=True)
+            adb.run()
+
+            # Do not remove, XHarness install seems to fail without an adb command called before the xharness command
+            getLogger().info("Preparing ADB")
+            adbpath = adb.stdout.strip()
+            try:
+                installCmd = [
+                    adbpath,
+                    'install',
+                    packagepath,
+                ]
+                
+                clearLogsCmd = [
+                    adbpath,
+                    'logcat',
+                    '-c'
+                ]
+
+                startInstrumentationCmd = [
+                    adbpath,
+                    'shell',
+                    'am',
+                    'instrument',
+                    '-w',
+                    f'{packagename}/{instrumentationname}'
+                ]
+                
+                printMauiLogsCmd = [
+                    adbpath,
+                    'shell',
+                    'logcat',
+                    '-d',
+                    'v',
+                    'tag',
+                    '-s',
+                    '"DOTNET,MAUI"'
+                ]
+
+                getLogger().info("Installing APK")
+                RunCommand(installCmd, verbose=True).run()
+
+                # Clear logs
+                getLogger().info("Clearing logs")
+                RunCommand(clearLogsCmd, verbose=True).run()
+
+                # Run instrumentation
+                getLogger().info("Running instrumentation")
+                RunCommand(startInstrumentationCmd, verbose=True).run()
+                
+                # Print logs
+                getLogger().info("Printing logs")
+                RunCommand(printMauiLogsCmd, verbose=True).run()
+                
+                ## Get logs off device and upload to helix (TODO: Make this optional, potentially add different methods of getting logs)
+                defaultDeviceBdnOutputDir = f'/sdcard/Android/data/{packagename}/files/'
+                pullFilesFromDeviceCmd = [
+                    adbpath,
+                    'pull',
+                    defaultDeviceBdnOutputDir,
+                    TRACEDIR
+                ]
+
+                RunCommand(pullFilesFromDeviceCmd, verbose=True).run()
+
+                # Replace the JSON with the correct values
```

Comment:
For my own understanding, why the _perf-lab-report_ does not have correct values? Why is i.e. `repo` or `branch` incorrect?

Reply:
I was unable to find a way to pass through the correct data to the built app at runtime such that it could be set as an environment variable. The environment variables are then used by the PerfLabExporter in the .json it generates, requiring all the values to have something set.

Reply:
Thanks.